### PR TITLE
feat(web): add types to dispatcher

### PR DESCRIPTION
**Pull request by martabal**
## Changes made in this PR

Continuing work on #5698, this PR adds types to every `createEventDispatcher`

### CODE REVIEW

File: web/src/lib/components/admin-page/delete-confirm-dialoge.svelte [hunk 1]
```diff
@@ -6,19 +6,22 @@
 
   export let user: UserResponseDto;
 
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    userDeleteSuccess: void;
```
Comment (jrasm91) [inline web/src/lib/components/admin-page/delete-confirm-dialoge.svelte:10]: can these just be `success` and `fail`?

File: web/src/lib/components/album-page/thumbnail-selection.svelte [hunk 2]
```diff
@@ -10,7 +10,10 @@
   export let album: AlbumResponseDto;
 
   let selectedThumbnail: AssetResponseDto | undefined;
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    close: void;
+    thumbnailSelected: { asset: AssetResponseDto | undefined };
```
Comment (jrasm91) [inline web/src/lib/components/album-page/thumbnail-selection.svelte:15]: Can this just be `thumbnail` and can the data just be AssetResponseDto | undefined?

File: web/src/lib/components/faces-page/merge-face-selector.svelte [hunk 3]
```diff
@@ -24,7 +24,10 @@
   let screenHeight: number;
   let isShowConfirmation = false;
 
-  let dispatch = createEventDispatcher();
+  let dispatch = createEventDispatcher<{
+    goBack: void;
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/merge-face-selector.svelte:28]: (back)

File: web/src/lib/components/faces-page/show-hide.svelte [hunk 4]
```diff
@@ -7,7 +7,12 @@
   import LoadingSpinner from '$lib/components/shared-components/loading-spinner.svelte';
   import { mdiClose, mdiEye, mdiEyeOff, mdiRestart } from '@mdi/js';
 
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    closeClick: void;
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/show-hide.svelte:11]: (close)

File: web/src/lib/components/faces-page/show-hide.svelte [hunk 5]
```diff
@@ -7,7 +7,12 @@
...
+    resetVisibility: void;
+    toggleVisibility: void;
+    doneClick: void;
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/show-hide.svelte:14]: (done)

File: web/src/lib/components/faces-page/show-hide.svelte [hunk 6]
```diff
@@ -7,7 +7,12 @@
   import LoadingSpinner from '$lib/components/shared-components/loading-spinner.svelte';
   import { mdiClose, mdiEye, mdiEyeOff, mdiRestart } from '@mdi/js';
 
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    closeClick: void;
+    resetVisibility: void;
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/show-hide.svelte:12]: (reset)

File: web/src/lib/components/faces-page/show-hide.svelte [hunk 7]
```diff
@@ -7,7 +7,12 @@
...
+    toggleVisibility: void;
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/show-hide.svelte:13]: (change)

File: web/src/lib/components/forms/create-user-form.svelte [hunk 8]
```diff
@@ -24,7 +24,10 @@
       canCreateUser = true;
     }
   }
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    userCreated: void;
```
Comment (jrasm91) [inline web/src/lib/components/forms/create-user-form.svelte:28]: (submit)

File: web/src/lib/components/shared-components/album-selection-modal.svelte [hunk 9]
```diff
@@ -13,15 +13,9 @@
...
-    newAlbum: {
-      albumName: string;
-    };
-    album: {
-      album: AlbumResponseDto;
-    };
-    newSharedAlbum: {
-      albumName: string;
-    };
+    newAlbum: string;
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/album-selection-modal.svelte:16]: (album, { shared: true })

File: web/src/lib/components/shared-components/control-app-bar.svelte [hunk 10]
```diff
@@ -13,7 +13,9 @@
 
   let appBarBorder = 'bg-immich-bg border border-transparent';
 
-  const dispatch = createEventDispatcher();
+  const dispatch = createEventDispatcher<{
+    closeButtonClick: void;
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/control-app-bar.svelte:17]: (close)

**Review by jrasm91 — COMMENTED** [review]
Thanks for working on this. Just a few renaming suggestions.

File: web/src/lib/components/shared-components/album-selection-modal.svelte (hunk 9 above)
Comment (martabal) [reply web/src/lib/components/shared-components/album-selection-modal.svelte:16]: I kept only `newAlbum` they both do the same thing

File: web/src/lib/components/user-settings-page/user-api-key-list.svelte [hunk 11]
```diff
@@ -112,7 +112,7 @@
 <section class="my-4">
   <div class="flex flex-col gap-2" in:fade={{ duration: 500 }}>
     <div class="mb-2 flex justify-end">
-      <Button size="sm" on:click={() => (newKey = { name: 'API Key' })}>New API Key</Button>
+      <Button size="sm" on:click={() => (newKey ? (newKey.name = 'API Key') : '')}>New API Key</Button>
```
Comment (jrasm91) [inline web/src/lib/components/user-settings-page/user-api-key-list.svelte:115]: what is this change?

Comment (martabal) [reply web/src/lib/components/user-settings-page/user-api-key-list.svelte]: Reverted. I wanted it not use `Partial<APIKeyResponseDto>` but it seems it's way easier to use it

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**